strawberryfields/backends/gaussianbackend/gaussiancircuit.py

```diff
@@ -400,9 +412,10 @@ def measure_dyne(self, covmat, indices):
 
         r = self.smean()
         (va, vc) = ops.chop_in_blocks_vector(r, expind)
-        vm = np.random.multivariate_normal(vc, C)
-
-        va = va+np.dot(np.dot(B, np.linalg.inv(C+covmat)), vm-vc)
+        vm = np.random.multivariate_normal(vc, C, size=shots)
+        # The next line is a hack in that it only updates conditioned on the first samples value
+        # should still wotk if shots = 1
+        va = va+np.dot(np.dot(B, np.linalg.inv(C+covmat)), vm[0]-vc)
```

Comment:
Conditional state updates are a tricky question when `shots>1`. I think the best solution is to only allow `shots` for measurements that appear at the end of a circuit, and in such cases, no conditional state is updated in the remaining modes.
A short term solution is to just raise an exception in any measurement other than `measure_fock` (which necessarily must appear at the end of a gaussian circuit)

Reply:
Is this:

![image](https://user-images.githubusercontent.com/991946/60272247-cbfd5c80-98c1-11e9-86d1-99e8a4e234fe.png)

the expected behaviour when you measure a subset of the modes, to return None

Reply:
yes